forge/csrc/passes/commute_utils.cpp

```diff
@@ -898,18 +898,213 @@ bool can_commute_past_op(
     return (is_elementwise(op) and op->op_name() != "interleave") or is_quantization_ops(op);
 }
 
+/**
+ * @brief Updates the attributes and named attributes of "select" operation with new values for dimension, begin,
+ * length, and stride.
+ */
+void update_select_attr(
+    graphlib::OpNode *select_op,
+    int select_dim,
+    std::optional<int> begin,
+    std::optional<int> length,
+    std::optional<int> stride)
+{
+    log_trace("op name = {}", select_op->name());
+
+    if (select_op->op_name() != "select")
+    {
+        return;
+    }
+    assert(select_op->op_name() == "select" && "update_select_attr called for a non-select operation");
```

Comment:
Should this assert be par of above if condition? With above if cond, we'll never hit this assert